src/gui/plotWidget.cpp

```diff
@@ -0,0 +1,242 @@
+ *                                                                            *
+ * You should have received a copy of the GNU General Public License          *
+ * along with linescaleGUI. If not, see <http://www.gnu.org/licenses/>.       *
+ ******************************************************************************/
+/**
+ * @file plotWidget.cpp
+ * @authors Gschwind, Weber, Schoch, Niederberger
+ */
+
+#include "plotWidget.h"
+#include <QDebug>
+
+Plot::Plot(QWidget* parent) : QWidget(parent) {
+    updateTimer = new QTimer(this);
+    updateTimer->setInterval(16);
+    updateTimer->setTimerType(Qt::TimerType::PreciseTimer);
+    connect(updateTimer, &QTimer::timeout, this, &Plot::updatePlot);
+
+    disableReplotTimer = new QTimer(this);
+    disableReplotTimer->setInterval(1000);
+    disableReplotTimer->setTimerType(Qt::TimerType::VeryCoarseTimer);
+    connect(disableReplotTimer, &QTimer::timeout, this, &Plot::disableUpdating);
+
+    customPlot = new QCustomPlot(parent);
+    // customPlot->setOpenGl(true); @todo Fix HiDPI bug in QCustomPlot library.
+    customPlot->setNoAntialiasingOnDrag(true);
+
+    customPlot->setInteractions(QCP::iRangeDrag | QCP::iRangeZoom | QCP::iSelectAxes |
+                                QCP::iSelectPlottables | QCP::iMultiSelect);
+    customPlot->xAxis->setLabel("");
+    customPlot->yAxis->setLabel("");
+    customPlot->legend->setVisible(false);
+
+    customPlot->addGraph();
+    customPlot->graph()->setLineStyle(QCPGraph::LineStyle::lsLine);
+
+    QPen graphPen;
+    graphPen.setColor(QColor(78, 50, 168, 255));
+    graphPen.setWidthF(1.5);
+    customPlot->graph()->setPen(graphPen);
+    customPlot->replot();
+    customPlot->rescaleAxes();
+
+    // connect slot that ties some axis selections together (especially opposite axes):
+    connect(customPlot, &QCustomPlot::selectionChangedByUser, this, &Plot::selectionChanged);
+    // connect slots that takes care that when an axis is selected, only that direction can be
+    // dragged and zoomed:
+    connect(customPlot, &QCustomPlot::mousePress, this, &Plot::mousePress);
+    connect(customPlot, &QCustomPlot::mouseWheel, this, &Plot::mouseWheel);
+    connect(customPlot, &QCustomPlot::mouseMove, this, &Plot::mouseMove);
+    customPlot->setContextMenuPolicy(Qt::ContextMenuPolicy::CustomContextMenu);
+    connect(customPlot, &QCustomPlot::customContextMenuRequested, this, &Plot::contextMenuRequest);
+
+    auto layout = new QVBoxLayout(this);
+    layout->setContentsMargins(0, 0, 0, 0);
+    layout->addWidget(customPlot);
+    this->setLayout(layout);
+    this->setContentsMargins(0, 0, 0, 0);
+
+    deleteGraphAction = new QAction("Delete graph", this);
+    deleteGraphAction->setShortcuts({QKeySequence::Delete, QKeySequence::Backspace});
+    connect(deleteGraphAction, &QAction::triggered, this, [=]() {
+        for (auto g : customPlot->selectedGraphs()) {
+            customPlot->removeGraph(g);
+        }
+    });
```

Comment:
The automatic scaling of the y-axis is not reset after deletion.
Thus a low value in a new plot is not visible.

=> Maybe reset the max and min values to their original values
=> Alternatively: add a context action to reset the plot

Reply:
> => Maybe reset the max and min values to their original values

Done.

Reply:
Reset works, if we call `updatePlot()` after the for loop, the plot will be updated regardless of new values.